python/pyspark/ml/clustering.py

```diff
@@ -167,6 +167,200 @@ def getInitSteps(self):
+            (term indices, term weights in topic).
+            Each topic's terms are sorted in order of decreasing weight.
+        """
+        return self._call_java("describeTopics", maxTermsPerTopic)
+
+
+class DistributedLDAModel(LDAModel):
+    """
+    Model fitted by LDA.
+
+    .. versionadded:: 1.7.0
+    """
+    def toLocal(self):
+        return self._call_java("toLocal")
+
+
+class LocalLDAModel(LDAModel):
+    """
+    Model fitted by LDA.
+
+    .. versionadded:: 1.7.0
+    """
+    pass
+
+
+class LDA(JavaEstimator, HasFeaturesCol, HasMaxIter, HasSeed, HasCheckpointInterval):
+    """ A clustering model derived from the LDA method.
+
+    Latent Dirichlet Allocation (LDA), a topic model designed for text documents.
+    Terminology
+    - "word" = "term": an element of the vocabulary
+    - "token": instance of a term appearing in a document
+    - "topic": multinomial distribution over words representing some concept
+    References:
+    - Original LDA paper (journal version):
+    Blei, Ng, and Jordan.  "Latent Dirichlet Allocation."  JMLR, 2003.
+
+    >>> from pyspark.mllib.linalg import Vectors, SparseVector
+    >>> from pyspark.ml.clustering import LDA
+    >>> df = sqlContext.createDataFrame([[1, Vectors.dense([0.0, 1.0])], \
+        [2, SparseVector(2, {0: 1.0})],], ["id", "features"])
+    >>> lda = LDA(k=2, seed=1, optimizer="em")
+    >>> model = lda.fit(df)
+    >>> model.isDistributed()
+    True
+    >>> localModel = model.toLocal()
+    >>> localModel.isDistributed()
+    False
+    >>> model.vocabSize()
+    2
+    >>> model.describeTopics().show()
+    +-----+-----------+--------------------+
+    |topic|termIndices|         termWeights|
+    +-----+-----------+--------------------+
+    |    0|     [1, 0]|[0.50401530077160...|
+    |    1|     [0, 1]|[0.50401530077160...|
+    +-----+-----------+--------------------+
+    ...
+    >>> model.topicsMatrix()
+    DenseMatrix(2, 2, [0.496, 0.504, 0.504, 0.496], 0)
+
+    .. versionadded:: 1.7.0
+    """
+
+    # a placeholder to make it appear in the generated doc
+    k = Param(Params._dummy(), "k", "number of clusters to create")
+    optimizer = Param(Params._dummy(), "optimizer", "LDA optimizer")
+
+    @keyword_only
+    def __init__(self, featuresCol="features", k=2,
+                 optimizer="online", learningOffset=5, learningDecay=0.51,
+                 subsamplingRate=0.05, optimizeDocConcentration=True,
+                 checkpointInterval=10, maxIter=20, seed=None):
+        """
+        __init__(self, featuresCol="features", predictionCol="prediction", k=2, \
+                 initMode="k-means||", initSteps=5, tol=1e-4, maxIter=20, seed=None)
+        """
+        super(LDA, self).__init__()
+        self._java_obj = self._new_java_obj("org.apache.spark.ml.clustering.LDA", self.uid)
+        self.k = Param(self, "k", "number of clusters to create")
+        self.optimizer = Param(self, "optimizer", "LDA optimizer")
```

Comment:
You should define ```learningOffset```, ```learningDecay```, ```subsamplingRate```, ```optimizeDocConcentration``` and their default values, then use can set these parameters.